src/App.tsx

```diff
@@ -5,7 +5,34 @@ import { FindMovie } from './components/FindMovie';
 import { Movie } from './types/Movie';
 
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [movies, setMovies] = useState<Movie[]>([]);
+
+  const onAdd = (movie: Movie) => {
+    setMovies((prevMovies) => {
+      if (!movie) {
+        return prevMovies;
+      }
+
+      const movieToAdd = {
+        ...movie,
+      };
+
+      const isMovieAdded = prevMovies.some((film => (
+        film.imdbId === movie.imdbId
+      )));
+
+      if (isMovieAdded) {
+        alert('the movie is already in your list');
```

Comment:
avoid `alerts`, use some value in state for error and conditional rendering below the query input